Physics/Dynamics/ProductionFormFactor.hpp

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) 2013, 2019 The ComPWA Team.
+// This file is part of the ComPWA framework, check
+// https://github.com/ComPWA/ComPWA/license.txt for details.
+
+#ifndef PHYSICS_DYNAMICS_PRODUCTION_FORMFACTOR_HPP_
+#define PHYSICS_DYNAMICS_PRODUCTION_FORMFACTOR_HPP_
+
+#include <boost/property_tree/ptree.hpp>
+#include <memory>
+#include <vector>
+
+#include "AbstractDynamicalFunction.hpp"
+#include "Core/Exceptions.hpp"
+#include "Core/Functions.hpp"
+#include "Core/Spin.hpp"
+#include "FormFactor.hpp"
+
+namespace ComPWA {
+namespace Physics {
+namespace Dynamics {
+
+/// The production formfactor is implemented in Blatt-Weisskopf type \f$B_{L}(q)\f$
+/// Ref. Dalitz Plot Analysis, PDG 2012, where 
+/// \f[
+/// \frac{B^{\prime}(q_{0})}{B^{\prime}_{L}(q)} = q^{L}B^{\prime}_{L}(q, q_{0})
+/// \f]
+/// \f$B_{L}(q)(i.e., F_L(q))$\f with \f$L$\f up to 4 can be found in BNL-QGS-06-101
+class ProductionFormFactor
+    : public ComPWA::Physics::Dynamics::AbstractDynamicalFunction {
+
+public:
+  //============ CONSTRUCTION ==================
+  ProductionFormFactor(std::string name, 
```

Comment:
With the decorator pattern you would pass a shared_ptr of the "undecorated" class here. That would be just another AbstractDynamicalFunction. So `std::shared_ptr<AbstractDynamicalFunction>`. Then you also define a member in this class which you assign this to. Finally you only need to modify the `evaluate()` function as shown below